src/Oro/Bundle/BatchBundle/ORM/QueryBuilder/CountQueryBuilderOptimizer.php

```diff
@@ -95,7 +95,7 @@ public function getCountQueryBuilder(QueryBuilder $originalQb)
             $this->addJoins($qb, $parts);
         }
         if (!$parts['groupBy']) {
-            $fieldsToSelect[] = $this->getFieldFQN($this->idFieldName);
+            $fieldsToSelect[] = sprintf('DISTINCT %s', $this->getIdFieldFQN());
```

Comment:
I suppose here can come cross join or table with composed primary key, so it looks like we shouldn't use distinct keyword